src/browser.js

```diff
@@ -10,10 +10,68 @@ exports.formatArgs = formatArgs;
 exports.save = save;
 exports.load = load;
 exports.useColors = useColors;
-exports.storage = 'undefined' != typeof chrome
-               && 'undefined' != typeof chrome.storage
-                  ? chrome.storage.local
-                  : localstorage();
+
+var isFirefox = false;
+var isFirefoxNewerThan31 = false;
+var isAppleWebkit = false;
+
+if (typeof navigator !== 'undefined' &&
+    typeof navigator.userAgent !== 'undefined') {
+  isFirefox = navigator.userAgent.toLowerCase().match(/firefox\/(\d+)/)
```

Comment:
Good news: I just tested the following in Firefox:

```javascript
> browser.storage.local.get('debug', console.log)
< undefined
< Object { debug: "*" }
```

In other words, you don't actually need any browser detection here, as Firefox supports the callback pattern **in addition** to the promise pattern.